keps/sig-storage/1979-object-storage-support/README.md

```diff
@@ -711,214 +647,324 @@ BucketAccess {
+COSI uses `Request.name` as an idempotency key. The driver should ensure that multiple DriverCreateBucket calls for the same name do not result in more than one OSP backend bucket being provisioned corresponding to that name. Using or appending random identifiers can lead to multiple unused buckets being created in the OSP backend in the event of timing-related driver/sidecar failures or restarts.
 
-## COSI Driver gRPC API
+The Sidecar uses the name of the Bucket resource as the input value for `Request.name`. This will be `bc-<BucketClaim.UID>` for dynamically-provisioned Buckets -- statistically likely to be globally unique even between multiple Kubernetes clusters.
 
-#### DriverGetInfo
+Input `parameters` are the opaque parameters copied from the Bucket (originating from BucketClass). Drivers can use these parameters to configure OSP bucket features based on the Administrator's BucketClass configuration.
 
-This gRPC call responds with the name of the driver. The name is used to identify the appropriate driver for a given BucketRequest or BucketClaim.
+The returned `bucketID` should be a unique identifier for the OSP bucket known to the driver. This value will be used by COSI to make all subsequent calls related to this bucket, so the driver must be able to correlate `bucketID` to the OSP backend bucket. It is easiest for drivers to use the request `name` field as both `bucketID` and as the OSP backend bucket identifier, but this is not strictly required.
 
-```
-    DriverGetInfo
-    |------------------------------------------|       |---------------------------------------|
-    | grpc DriverGetInfoRequest{}              | ===>  | DriverGetInfoResponse{                |
-    |------------------------------------------|       |   "name": "s3.amazonaws.com"          |
-                                                       | }                                     |
-                                                       |---------------------------------------|
-```
+```go
+DriverCreateBucket(DriverCreateBucketRequest) DriverCreateBucketResponse
 
-#### DriverCreateBucket
+DriverCreateBucketRequest{
+  "name": "<Bucket.name>", // will be "bc-<BucketClaim.UID>" for dynamically-provisioned Buckets
 
-This gRPC call creates a bucket in the OSP, and returns information about the new bucket. This api must be idempotent. The input to this call is the name of the bucket and an opaque parameters field.
+  "parameters": { // copied from Bucket.parameters
+    "<key>": "<value>"
+    // ...
+  }
+}
 
-The returned `bucketID` should be a unique identifier for the bucket in the OSP. This value could be the name of the bucket too. This value will be used by COSI to make all subsequent calls related to this bucket.
+DriverCreateBucketResponse{
+  Bucket bucket
+}
+
+Bucket {
+  "bucketID": "<ID returned by driver>" // will be applied to Bucket.status.bucketID
+  "protocols": { // all protocols supported by the bucket, with bucket info
```

Comment:
Note: this must contain at least one element, otherwise provisioning should be considerd failed.

Reply:
Added the note here and to credentials below